File: d66-social/src/main/java/com/eroldmr/d66/subreddit/vote/VoteRepository.java
package com.eroldmr.d66.subreddit.vote;

import com.eroldmr.d66.appuser.AppUser;
import com.eroldmr.d66.subreddit.post.Post;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface VoteRepository extends JpaRepository<Vote, Long> {

  Optional<Vote> findByPostAndAppUser(Post post, AppUser appUser);

}

File: d66-social/src/main/kotlin/com/eroldme/d66/subredit/vote/VoteRepository.kt
package com.eroldme.d66.subredit.vote

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

/**
 * @author Mompati 'Patco' Keetile
 * @created 06-12-2022 @ 14:10
 */
@Repository
interface VoteRepository : JpaRepository<Vote, Long>
